### PR TITLE
ODS/XLSX: add FIELD_TYPES and HEADERS open options (fixes #8028)

### LINKED ISSUE #8028: OGR_{XLSX,ODS}_HEADERS and OGR_{XLSX,ODS}_FIELD_TYPES should be open options
## Expected behavior and actual behavior.

I've long wondered why the OGR_{XLSX,ODS}_HEADERS and OGR_{XLSX,ODS}_FIELD_TYPES options were _configuration_ options instead of _open_ options. Is there a reason for this or is it purely a legacy thing that could be addressed?

One of the side effect of not having the as open options is visible in QGIS where the data source manager's add [ogr] vector panel fails to display these quite useful options:

image

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

**Comment by rouault** [comment]
@dbaston  any idea how I should handle the definition option of identically named open option in 2 drivers ? I currently get a warning (turned as an error):
```
/__w/gdal/gdal/doc/source/drivers/vector/ods.rst:41: WARNING: Duplicate definition of HEADERS (previously defined at drivers/vector/xlsx:49)
/__w/gdal/gdal/doc/source/drivers/vector/ods.rst:52: WARNING: Duplicate definition of FIELD_TYPES (previously defined at drivers/vector/xlsx:60)
```

File: doc/source/drivers/vector/ods.rst [hunk 1]
```diff
@@ -31,6 +31,32 @@ Driver capabilities
...
 .. supports_virtualio::
 
+Open options
+------------
+
+.. versionadded:: 3.8
+
+The following open options are available:
+
+-  .. config:: HEADERS
```
Comment (dbaston) [inline doc/source/drivers/vector/ods.rst:41]:
```suggestion
-  .. oo:: HEADERS
```

File: doc/source/drivers/vector/ods.rst [hunk 2]
```diff
@@ -31,6 +31,32 @@ Driver capabilities
...
+      :default: AUTO
+
+      By default, the driver
+      will read the first lines of each sheet to detect if the first line
+      might be the name of columns. If set to FORCE, the driver will
+      consider the first line as the header line. If set to
+      DISABLE, it will be considered as the first feature. Otherwise
+      auto-detection will occur.
+
+-  .. config:: FIELD_TYPES
```
Comment (dbaston) [inline doc/source/drivers/vector/ods.rst:52]:
```suggestion
-  .. oo:: FIELD_TYPES
```

File: doc/source/drivers/vector/xlsx.rst [hunk 3]
```diff
@@ -39,6 +39,33 @@ Driver capabilities
...
 .. supports_virtualio::
 
+Open options
+------------
+
+.. versionadded:: 3.8
+
+The following open options are available:
+
+-  .. config:: HEADERS
```
Comment (dbaston) [inline doc/source/drivers/vector/xlsx.rst:49]:
```suggestion
-  .. oo:: HEADERS
```

File: doc/source/drivers/vector/xlsx.rst [hunk 4]
```diff
@@ -39,6 +39,33 @@ Driver capabilities
...
+      :default: AUTO
+
+      By default, the driver
+      will read the first lines of each sheet to detect if the first line
+      might be the name of columns. If set to FORCE, the driver will
+      consider the first line as the header line. If set to
+      DISABLE, it will be considered as the first feature. Otherwise
+      auto-detection will occur.
+
+-  .. config:: FIELD_TYPES
```
Comment (dbaston) [inline doc/source/drivers/vector/xlsx.rst:60]:
```suggestion
-  .. oo:: FIELD_TYPES
```

File: doc/source/drivers/vector/xlsx.rst [hunk 5]
```diff
@@ -39,6 +39,33 @@ Driver capabilities
...
+Open options
+------------
+
+.. versionadded:: 3.8
+
+The following open options are available:
+
+-  .. config:: HEADERS
+      :choices: FORCE, DISABLE, AUTO
+      :default: AUTO
```
Comment (dbaston) [inline doc/source/drivers/vector/xlsx.rst:51]:
```suggestion
      :default: AUTO
      :since: 3.8
```

File: doc/source/drivers/vector/xlsx.rst [hunk 6]
```diff
@@ -39,6 +39,33 @@ Driver capabilities
...
+      By default, the driver
+      will read the first lines of each sheet to detect if the first line
+      might be the name of columns. If set to FORCE, the driver will
+      consider the first line as the header line. If set to
+      DISABLE, it will be considered as the first feature. Otherwise
+      auto-detection will occur.
+
+-  .. config:: FIELD_TYPES
+      :choices: STRING, AUTO
+      :default: AUTO
```
Comment (dbaston) [inline doc/source/drivers/vector/xlsx.rst:62]:
```suggestion
      :default: AUTO
      :since: 3.8
```

File: doc/source/drivers/vector/ods.rst [hunk 7]
```diff
@@ -31,6 +31,32 @@ Driver capabilities
...
+Open options
+------------
+
+.. versionadded:: 3.8
+
+The following open options are available:
+
+-  .. config:: HEADERS
+      :choices: FORCE, DISABLE, AUTO
+      :default: AUTO
```
Comment (dbaston) [inline doc/source/drivers/vector/ods.rst:43]:
```suggestion
      :default: AUTO
      :since: 3.8
```

File: doc/source/drivers/vector/ods.rst [hunk 8]
```diff
@@ -31,6 +31,32 @@ Driver capabilities
...
+      By default, the driver
+      will read the first lines of each sheet to detect if the first line
+      might be the name of columns. If set to FORCE, the driver will
+      consider the first line as the header line. If set to
+      DISABLE, it will be considered as the first feature. Otherwise
+      auto-detection will occur.
+
+-  .. config:: FIELD_TYPES
+      :choices: STRING, AUTO
+      :default: AUTO
```
Comment (dbaston) [inline doc/source/drivers/vector/ods.rst:54]:
```suggestion
      :default: AUTO
      :since: 3.8
```

**Comment by dbaston** [comment]
>  identically named open option

Open options (declared with `oo::`) only need to be unique on a page, but configuration options (`config::`) need to be globally unique.

**Comment by rouault** [comment]
> Open options (declared with `oo::`) only need to be unique on a page, but configuration options (`config::`) need to be globally unique.

doh, sorry for the stupid question! I should have checked what I had written :-)

**Pull request merged**